src/overlays/actors/ovl_Dm_Opstage/z_dm_opstage.c

```diff
@@ -32,26 +31,108 @@ const ActorInit Dm_Opstage_InitVars = {
-static InitChainEntry D_80A9FD30[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneScale, 300, ICHAIN_STOP),
 };
 
-#endif
-
-extern InitChainEntry D_80A9FD30[];
-
-extern UNK_TYPE D_06000978;
-extern UNK_TYPE D_06001C98;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Opstage/DmOpstage_SetupAction.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Opstage/DmOpstage_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Opstage/DmOpstage_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Opstage/func_80A9FA58.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Opstage/DmOpstage_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Opstage/DmOpstage_Draw.s")
+void DmOpstage_SetupAction(DmOpstage* this, DmOpstageActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void DmOpstage_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 params;
+    DmOpstage* this = THIS;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DmOpstage_SetupAction(this, func_80A9FA58);
+    Actor_SetScale(&this->dyna.actor, 0.1f);
+    params = DMOPSTAGE_GET_TYPE(thisx);
+    if (params == 0) {
+        DynaPolyActor_Init(&this->dyna, 0);
+        DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_keikoku_demo_Colheader_001C98);
+        params = DMOPSTAGE_GET_TYPE(thisx);
+    }
+    if (params > 0) {
+        this->posX = this->dyna.actor.world.pos.x;
+        this->posY = this->dyna.actor.world.pos.y;
+        this->posZ = this->dyna.actor.world.pos.z;
+        this->dyna.actor.world.pos.x = 0.0f;
+        this->dyna.actor.world.pos.y = 0.0f;
+        this->dyna.actor.world.pos.z = 0.0f;
+    }
+}
+
+void DmOpstage_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    DmOpstage* this = THIS;
+
+    if (DMOPSTAGE_GET_TYPE(thisx) == 0) {
+        DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+void func_80A9FA58(DmOpstage* this, GlobalContext* globalCtx) {
+    s16 params;
+    s32 actionIndex;
+
+    params = this->dyna.actor.params;
+    if ((params & 0xFF) == 0) {
+        if (Cutscene_CheckActorAction(globalCtx, 115)) {
+            actionIndex = Cutscene_GetActorActionIndex(globalCtx, 115);
+            if (globalCtx->csCtx.actorActions[actionIndex]->action == 2) {
+                this->dyna.actor.scale.x = 0.075f;
+                this->dyna.actor.scale.z = 0.3f;
+            } else {
+                this->dyna.actor.scale.x = 0.1f;
+                this->dyna.actor.scale.z = 0.1f;
+            }
+            Cutscene_ActorTranslateAndYaw(&this->dyna.actor, globalCtx, actionIndex);
+        }
+    } else if (Cutscene_CheckActorAction(globalCtx, (((params >> 8) & 0xFF) + 0x74) & 0xFFFF)) {
+        Cutscene_ActorTranslateAndYaw(
+            &this->dyna.actor, globalCtx,
+            Cutscene_GetActorActionIndex(globalCtx, (((this->dyna.actor.params >> 8) & 0xFF) + 0x74) & 0xFFFF));
+    }
+}
```

Comment:
```suggestion
void func_80A9FA58(DmOpstage* this, GlobalContext* globalCtx) {
    s32 actionIndex;

    if (DMOPSTAGE_GET_TYPE(&this->dyna.actor) == 0) {
        if (Cutscene_CheckActorAction(globalCtx, 115)) {
            actionIndex = Cutscene_GetActorActionIndex(globalCtx, 115);
            if (globalCtx->csCtx.actorActions[actionIndex]->action == 2) {
                this->dyna.actor.scale.x = 0.075f;
                this->dyna.actor.scale.z = 0.3f;
            } else {
                this->dyna.actor.scale.x = 0.1f;
                this->dyna.actor.scale.z = 0.1f;
            }
            Cutscene_ActorTranslateAndYaw(&this->dyna.actor, globalCtx, actionIndex);
        }
    } else if (Cutscene_CheckActorAction(globalCtx, ((this->dyna.actor.params >> 8) & 0xFF) + 0x74)) {
        Cutscene_ActorTranslateAndYaw(
            &this->dyna.actor, globalCtx,
            Cutscene_GetActorActionIndex(globalCtx, ((this->dyna.actor.params >> 8) & 0xFF) + 0x74));
    }
}
```
One extra thing to note besides the removal of the params temp, the `& 0xFFFF`s could also be removed as the function args take `u16`s so Ido automatically does it.

Reply:
Also going to need to make a macro for the `((this->dyna.actor.params >> 8) & 0xFF)`. I would say leave the `+ 0x74` out of the macro for now.